### DISCUSSION TITLE
setTextDatum and printToSprite

### DISCUSSION THREAD

**Discussion by straytowel** [Bodmer/TFT_eSPI#1861] [Q&A - Sprites] [answered]
Is text alignment possible with printToSprite?

Basic example taken from a slightly modified Font_Demo_4_Array

```
 for (int i = 0; i <= 110; i++) {
    tft.setCursor(110, 110);
    spr.setTextDatum(MR_DATUM);
    spr.printToSprite((String) i ); 
    delay(60);
  }
```

I've tried creating the sprite first and printing to it as well with no luck either. Always stuck at left top justified.
I can get this to work with regular print, drawNumber or drawFloat, just not with printToSprite. Or is there something I'm missing?

**Comment by Bodmer** [comment, marked as answer]
printToSprite was intended to simplify the use of test rendering via sprites. If the sprite does not exist then it will be created as a minimal size to hold the text, in this case the datum use is not relevant. In a psrite that has slready been created the printToSprite function can only be used to print at a cursor position and datum is not used. See Font_Demo_4_xxxx for details of use.
